test/controllers/api/v2/activation_keys_controller_test.rb

```diff
@@ -246,5 +246,12 @@ def test_remove_subscriptions_protected
         post(:remove_subscriptions, :organization_id => @organization.id, :id => @activation_key.id, :subscription_id => 123)
       end
     end
+
+    def test_remove_host_collections
+      @activation_key.update_attributes(host_collections: HostCollection.all)
+      assert(@activation_key.host_collection_ids.any?)
+      put(:remove_host_collections, organization_id: @organization.id, id: @activation_key.id, host_collection_ids: [])
+      assert_empty(@activation_key.host_collection_ids)
```

Comment:
This test is bad because it makes associated records in the DB, hits the API, then checks for no associated records; however, this bug was caused by the interaction of strong_params and deep_munge, which happens only on a web request. I'd love some help figuring out how to make this test less compute- and disk-intensive.